0_intro/exercises/fibonacci/src/lib.rs

```diff
@@ -1,5 +1,17 @@
 pub fn fibo(v: u32) -> u32 {
-    todo!()
+    if v == 0 {
+        0
+    }
+    else {
+        let mut f1 = 0;
+        let mut f2 = 1;
+        for _ in 1..v {
+            let tmp = f2;
+            f2 = f1 + f2;
+            f1 = tmp;
+        }
+        f2
+    }
```

Comment:
If you want to return early from a function, it's better to use the `return` keyword directly (that's the right way to use it in Rust, actually). This way, you don't have a lot
of code in an else or an if.
```suggestion
    if v == 0 {
        return 0;
    }
    
        let mut f1 = 0;
        let mut f2 = 1;
        for _ in 1..v {
            let tmp = f2;
            f2 = f1 + f2;
            f1 = tmp;
        }
        f2
```

Another way to do that, while keeping your structure, would be to separate functions more, and maybe define an inner function. This is the kind of programming that you will find in functional languages

```suggestion
    pub fn fibo(v: u32) -> u32 {
        fn fibo_non_null(v: u32) -> u32 {
        let mut f1 = 0;
        let mut f2 = 1;
        for _ in 1..v {
            let tmp = f2;
            f2 = f1 + f2;
            f1 = tmp;
        }
        f2
        }

    if v == 0 {
        0
    }
    else {
        fibo_non_null(v)
    }
}
```